etl/assets.py

```diff
@@ -148,17 +153,43 @@ def wikipedia_anti_recommendations_json_file(
         )
 
 
-@asset
-def wikipedia_arkg(
-    wikipedia_anti_recommendations: AntiRecommendationGraphTuple,
-    arkg_config: ArkgConfig,
-    output_config: OutputConfig,
-) -> None:
-    """Materialize a Wikipedia Anti-Recommendation Knowledge Graph asset."""
+def wikipedia_arkg_factory(
+    rdf_serialization_name: RdfSerializationName,
+    rdf_mime_type: RdfMimeType,
+    rdf_file_extension: RdfFileExtension,
+) -> AssetsDefinition:
+    """
+    A factory to build Wikipedia ARKG assets.
+
+    Multiple assets are created based on different RDF serialization types.
+    Each RDF serialization type is characterized by a (name, mime_type, file_extension) triple.
+    """
+
+    @asset(name=f"wikipedia_arkg_with_{rdf_serialization_name}_serialization")
+    def wikipedia_arkg(
+        output_config: OutputConfig,
+        wikipedia_anti_recommendations: AntiRecommendationGraphTuple,
+    ) -> ArkgDatabase.Descriptor:
+        """Materialize a Wikipedia Anti-Recommendation Knowledge Graph asset."""
+
+        parsed_output_config = output_config.parse()
+
+        wikipedia_arkg_database = ArkgDatabase.create(
+            requests_cache_directory=parsed_output_config.requests_cache_directory_path,
+            anti_recommendation_graphs=wikipedia_anti_recommendations,
+            arkg_file_path=parsed_output_config.wikipedia_arkg_file_path.with_suffix(
+                rdf_file_extension
+            ),
+        )
 
-    ArkgBuilderPipeline(
-        requests_cache_directory=output_config.parse().requests_cache_directory_path
-    ).construct_graph(wikipedia_anti_recommendations.anti_recommendation_graphs).dump(
-        output=output_config.parse().wikipedia_arkg_file_path,
-        mime_type=arkg_config.rdf_mime_type,
-    )
+        wikipedia_arkg_database.write(arkg_mime_type=rdf_mime_type)
```

Comment:
Nit: you'd usually call this `.dump` or `.save`. Oxigraph uses the former.

Also, passing the file path of the dump in the constructor and then calling a method with an additional dump parameter here seems weird. Dump usually takes (file path, format).